packages/extension/src/contexts/NetworkContext/NetworkContext.tsx

```diff
@@ -77,13 +84,14 @@ const NetworkProvider: FC = ({ children }) => {
   };
 
   const switchNetwork = useCallback(
-    async (network: Network) => {
+    async (network: Network, customNetworkName?: string, customNetworkServer?: string) => {
       try {
         await client?.disconnect();
-        const ws = new Client(NETWORK[network].server);
+        // If a server URL is provided, use it. Otherwise, use the pre-defined server for the given network
```

Comment:
Is there a case where: customNetworkServer is not provided? Shouldn't it be mandatory?

Reply:
It's not provided when the user connects to one of the predefined servers

Reply:
Oh yeah, right, good point!